src/components/dialog/dialog.component.js

```diff
@@ -273,13 +265,10 @@ Dialog.propTypes = {
   size: PropTypes.string,
   /** Determines if the close icon is shown */
   showCloseIcon: PropTypes.bool,
-  /** If true then the dialog receives focus when it opens */
-  autoFocus: PropTypes.bool,
   stickyFormFooter: PropTypes.bool
 };
 
 Dialog.defaultProps = {
-  autoFocus: true,
```

Comment:
If people used to have `autoFocus=false` this would no longer work. It's important that we communicate this in a `BREAKING CHANGE` commit.